python/src/catalog.rs

```diff
@@ -0,0 +1,123 @@
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+use std::collections::HashSet;
+use std::sync::Arc;
+
+use pyo3::exceptions::PyKeyError;
+use pyo3::prelude::*;
+// use pyo3::{PyIterProtocol, PyMappingProtocol};
+
+use datafusion::{
+    arrow::pyarrow::PyArrowConvert,
+    catalog::{catalog::CatalogProvider, schema::SchemaProvider},
+    datasource::{TableProvider, TableType},
+};
+
+#[pyclass(name = "Catalog", subclass)]
+pub(crate) struct PyCatalog {
+    catalog: Arc<dyn CatalogProvider>,
+}
+
+#[pyclass(name = "Database", subclass)]
```

Comment:
same here; this is `SchemaProvider` in DataFusion, right?

Reply:
Right. I'm going to remove this module from the PR to avoid confusion.